platform/cxxsupport/mstd_span

```diff
@@ -0,0 +1,406 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MSTD_SPAN_
+#define MSTD_SPAN_
+
+/* <mstd_span>
+ *
+ * - Includes toolchain's <span> if available
+ * - Otherwise provides an implementation of a C++20 equivalent std::span
+ *   - deduction guides available from C++17 and on
+ *   - provides nonstandard mstd::make_span functions to allow deduction pre C++17
```

Comment:
Comment's good, but it's just highlighted an implementation problem I didn't spot.

We would need the `mstd::make_span` functions even if the toolchain has `<span>` and C++17, for backwards compatibility.

Reply:
You've fixed the code, but now the comment isn't correct, with the "Otherwise...".

Reply:
You're right. I also added a note saying the make_span functions will remain after c++17 for extra clarification